packages/vue-client/src/views/GameView.vue

```diff
@@ -155,25 +147,41 @@ function makeMove(move_str: string) {
   requests.post(`/games/${props.gameId}/move`, move).catch(alert);
 }
 
+// subscribe to seat changes
+watchEffect((onCleanup) => {
+  const seatsMessage = `game/${props.gameId}/seats`;
+  socket.on(seatsMessage, (data) => {
+    players.value = data;
+  });
+  onCleanup(() => {
+    socket.off(seatsMessage);
+  });
+});
+
+// subscribe to game updates for specific seat
 watchEffect((onCleanup) => {
   if (!props.gameId) {
     return;
   }
-  const topic = `game/${props.gameId}`;
+  const topic =
+    playing_as.value == null
+      ? `game/${props.gameId}`
+      : `game/${props.gameId}/${playing_as.value}`;
   socket.emit("subscribe", [topic]);
-  socket.on("move", (state: GameStateResponse) => {
-    setNewState(state);
+
+  onCleanup(() => {
+    socket.emit("unsubscribe", [topic]);
   });
+});
 
-  const seatsMessage = `game/${props.gameId}/seats`;
-  socket.on(seatsMessage, (data) => {
-    players.value = data;
+// subscribe to state events
+watchEffect((onCleanup) => {
+  socket.on("move", (state: GameStateResponse) => {
+    setNewState(state);
```

Comment:
I'm thinking it's always ok to cache the state in the state_map. The displayed game state is only updated if round, seat, displayed_round, playing_as are in agreement.

Reply:
The observer subscription can also be handled in this way. When we receive the observer state while having a seat selected, we save it in state_map without updating the state displayed on the board. At least that is what I was trying to do with the first implementation.

Reply:
Ohh I see, I didn't realize this was already handled by the state_map.  Okay, so I guess this PR is probably not fixing the bug given that it's not the observer subscription.  I'm stumped!

Reply:
> in that case, should we go ahead and move forward with https://github.com/govariantsteam/govariants/pull/294 just to see if it fixes?

I think there is a better chance that this pull request fixes it 😅

Reply:
I just found I can reproduce (at least parts of) the issue consistently. I'll test this PR out later tonight!

Otherwise, I think I'll at least have something to debug with

Reply:
Thank you for looking into the problem, and please let me know if I can help you!

Reply:
I believe I was able to reproduce locally, and it seems that this pull request fixes it.

Reply:
Thanks for checking it out! I'm finding the same here. I don't understand which part of this fixes it tbh, but I can go ahead and merge.